Added language to pages docs to explain caveat on custom domains (#1068)
cloudflare/cloudflare-docs: products/pages/src/content/getting-started/index.md

Click the "Set up a domain" button, and provide the domain that you'd like to serve your Cloudflare Pages site on. Depending on the domain, you'll need to configure nameservers or add a custom CNAME record to point the domain to your Cloudflare Pages site.

> Please note that any custom domains (or subdomains) you wish to resolve to your Cloudflare Pages site need to be added via the previous "Set up a domain" process. Manually adding a custom CNAME record pointing to your Cloudflare Pages site without first associating the domain (or subdomains) in the Cloudflare Pages dashboard will result in your domain failing to resolve at the CNAME record address and display a 522 error.

Reviewer: This comment should go under the `#### Add a custom CNAME record` section. 

Slight nit in wording:
```
Please note that any custom domains (or subdomains) you wish to resolve to your Cloudflare Pages site need to be added via the previous "Set up a domain" process.
``` 
->
```
To ensure a custom domain is added successfully, you must go through the "Set up a custom domain" process described above".
```

Author: Thanks for the feedback @rita3ko, I've made the changes!

If you don't want to point your nameservers to Cloudflare, you can still use a custom CNAME record to use a single domain with Cloudflare Pages.

> To ensure a custom domain is added successfully, you must go through the "Set up a custom domain" process described above". Manually adding a custom CNAME record pointing to your Cloudflare Pages site without first associating the domain (or subdomains) in the Cloudflare Pages dashboard will result in your domain failing to resolve at the CNAME record address and display a 522 error.

Reviewer 2: ```suggestion
To ensure a custom domain is added successfully, you must go through the "Set up a custom domain" process described above. Manually adding a custom CNAME record pointing to your Cloudflare Pages site - without first associating the domain (or subdomains) in the Cloudflare Pages dashboard - will result in your domain failing to resolve at the CNAME record address, and display a 522 error.
```

> To ensure a custom domain is added successfully, you must go through the "Set up a custom domain" process described above. Manually adding a custom CNAME record pointing to your Cloudflare Pages site - without first associating the domain (or subdomains) in the Cloudflare Pages dashboard - will result in your domain failing to resolve at the CNAME record address, and display a 522 error.

Reviewer 3: ```suggestion
To ensure a custom domain is added successfully, you must go through the [Adding a custom domain](/getting-started#adding-a-custom-domain) process described above. Manually adding a custom CNAME record pointing to your Cloudflare Pages site - without first associating the domain (or subdomains) in the Cloudflare Pages dashboard - will result in your domain failing to resolve at the CNAME record address, and display a 522 error.
```

If you do not want to point your nameservers to Cloudflare, you can still use a custom CNAME record to use a single domain with Cloudflare Pages. You can configure your DNS settings in the **DNS** section of the Cloudflare dashboard. 

> To ensure a custom domain is added successfully, you must go through the [Adding a custom domain](/getting-started#adding-a-custom-domain) process described above. Manually adding a custom CNAME record pointing to your Cloudflare Pages site - without first associating the domain (or subdomains) in the Cloudflare Pages dashboard - will result in your domain failing to resolve at the CNAME record address, and display a [522 error](https://support.cloudflare.com/hc/en-us/articles/115003011431-Troubleshooting-Cloudflare-5XX-errors#522error).

Reviewer: Hey! Thanks for this PR. Could you wrap this in an <Aside>, and put it below the CNAME record, rather than above it? Then happy to merge.

Author: Hey, I've moved it under the CNAME block, but I'm not sure what you want me to wrap it in - the comment cuts off

Reviewer: Oh, I didn't realize it had cut off. An `<Aside>` (https://developers.cloudflare.com/docs-engine/reference/markdown#asides)

Author: Hi @rita3ko I have made these changes now. Thanks for the feedback!

| ------------ | ------------------ | ------------------ |
| `CNAME`      | `domain.com`       | `custom.pages.dev` |

> To ensure a custom domain is added successfully, you must go through the [Adding a custom domain](/getting-started#adding-a-custom-domain) process described above. Manually adding a custom CNAME record pointing to your Cloudflare Pages site - without first associating the domain (or subdomains) in the Cloudflare Pages dashboard - will result in your domain failing to resolve at the CNAME record address, and display a [522 error](https://support.cloudflare.com/hc/en-us/articles/115003011431-Troubleshooting-Cloudflare-5XX-errors#522error).

Reviewer 3: ```suggestion
<Aside>
To ensure a custom domain is added successfully, you must go through the [Adding a custom domain](/getting-started#adding-a-custom-domain) process described above. Manually adding a custom CNAME record pointing to your Cloudflare Pages site - without first associating the domain (or subdomains) in the Cloudflare Pages dashboard - will result in your domain failing to resolve at the CNAME record address, and display a [522 error](https://support.cloudflare.com/hc/en-us/articles/115003011431-Troubleshooting-Cloudflare-5XX-errors#522error).
</Aside>
```

Reviewer 3: Per @rita3ko 's suggestion
